packages/builder/src/pages/builder/workspace/[application]/agent/[agentId]/AgentToolConfigModal.svelte

```diff
@@ -2,94 +2,143 @@
   import BudibaseLogo from "../logos/Budibase.svelte"
-  import ConfluenceLogo from "../logos/Confluence.svelte"
-  import GithubLogo from "../logos/Github.svelte"
+
+  interface ToolSourceOption {
+    name: string
+    type: ToolSourceType
+    description: string
+  }
+  export let agentId: string
 
   const Logos: Record<string, ComponentType> = {
     BUDIBASE: BudibaseLogo,
-    CONFLUENCE: ConfluenceLogo,
-    GITHUB: GithubLogo,
-    BAMBOOHR: BambooHRLogo,
   }
 
   const ToolSources = [
     {
       name: "Budibase",
-      type: "BUDIBASE",
+      type: ToolSourceType.BUDIBASE,
       description: "Connect agent to your Budibase tools",
     },
     {
-      name: "Github",
-      type: "GITHUB",
-      description: "Automate development workflows.",
-    },
-    {
-      name: "Confluence",
-      type: "CONFLUENCE",
-      description: "Connect agent to your teams documentation",
-    },
-    {
-      name: "Sharepoint",
-      type: "SHAREPOINT",
-      description: "Sharepoint stuff",
-    },
-    {
-      name: "JIRA Service Manaagement",
-      type: "JIRA_SM",
-      description: "Automate ITSM Workflows",
-    },
-    {
-      name: "BambooHR",
-      type: "BAMBOOHR",
-      description: "Automate HR workflows and employee management",
+      name: "REST API",
+      type: ToolSourceType.REST_QUERY,
+      description: "Use queries from your REST APIs as agent tools",
     },
   ]
 
-  export let agentId: string
-
   const dispatch = createEventDispatcher()
 
   let modal: Modal
   let mode: "select" | "configure" = "select"
-  let selectedSourceType: any = null
+  let selectedSourceType: ToolSourceOption | null = null
   let editingSource: AgentToolSourceWithTools | null = null
 
-  let config: Record<string, string> = {}
+  let guidelines = ""
   let disabledTools: string[] = []
-  let toolsList: any[] = []
+  let toolsList: Tool[] = []
+  let label = ""
+
+  let selectedDatasourceId: string = ""
+  let selectedQueryIds: string[] = []
+
+  $: restDatasources = ($datasources.list || []).filter(
+    ds => ds.source === IntegrationTypes.REST
+  )
+
+  // We only allow one Budibase tool source per agent
+  $: hasBudibaseSource = $agentsStore.toolSources.some(
+    ts => ts.type === ToolSourceType.BUDIBASE
+  )
+  $: availableToolSources = ToolSources.filter(
+    source => !(source.type === ToolSourceType.BUDIBASE && hasBudibaseSource)
+  )
+
+  $: datasourceOptions = restDatasources.map(ds => ({
+    label: ds.name || ds._id!,
+    value: ds._id!,
+  }))
+
+  $: queriesForDatasource = ($queries.list || []).filter(
+    (q: Query) => q.datasourceId === selectedDatasourceId
+  )
+
+  // Auto-populate label and queries when datasource changes
+  let previousDatasourceId = ""
+  $: if (
+    selectedDatasourceId &&
+    selectedDatasourceId !== previousDatasourceId
+  ) {
+    const isInitialLoad = !previousDatasourceId && editingSource
+    previousDatasourceId = selectedDatasourceId
+
+    if (!isInitialLoad) {
```

Comment:
NAB: Prefer positive booleans, so instead of `!isInitialLoad` do `hasLoaded` instead